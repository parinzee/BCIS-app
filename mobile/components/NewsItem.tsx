import { View, StyleSheet } from "react-native";
import { Caption, Surface, Title } from "react-native-paper";
import Markdown from "react-native-markdown-display";
import MarkdownRenderRules from "../constants/MarkdownRenderRules";

interface NewsItemProps {
  emoji: string;
  title: string;
  description: string;
  dateUpdated: Date;
  width: number;
}

export default function NewsItem({
  emoji,
  title,
  description,
  dateUpdated,
  width,
}: NewsItemProps) {
  return (
    <Surface
      style={{
        ...styles.surface,
        width: width,
      }}
    >
      <View style={styles.titleBar}>
        <Title>{title}</Title>
        <Title>{emoji}</Title>
      </View>
      <View style={styles.body}>
        <Markdown rules={MarkdownRenderRules} style={styles}>
          {description}
        </Markdown>
        <Caption style={styles.datePosted}>
          Posted{" "}
          {`${dateUpdated.getDate()}/${dateUpdated.getMonth()}/${dateUpdated.getFullYear()}`}
        </Caption>
      </View>
    </Surface>
  );
}

const styles = StyleSheet.create({
  titleBar: {
    flexDirection: "row",
    justifyContent: "space-between",
    borderBottomWidth: 1.25,
    borderBottomColor: "#7e7e7e",
  },
  surface: {
    marginVertical: 20,
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderRadius: 10,
    elevation: 4,
    height: 150,
    maxWidth: 700,
    alignSelf: "center",
  },
  surfaceLarge: {
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderRadius: 10,
    elevation: 4,
    height: 150,
  },
  body: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "space-between",
  },
  datePosted: {
    flex: 0,
    textAlign: "right",
  },
  link: {
    color: "#4286f4",
  },
});
