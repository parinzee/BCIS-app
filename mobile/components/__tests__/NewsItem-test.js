import * as React from "react";
import renderer from "react-test-renderer";
import NewsItem from "../NewsItem";

it(`renders correctly`, () => {
  const tree = renderer
    .create(
      <NewsItem
        title="Amazing Title"
        emoji="😁"
        description="Test some **bold**, *italic*, and **both** and [links](https://github.com/parinzee)!"
        dateUpdated={new Date("2022-07-17T15:22:32Z")}
        width={400}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
