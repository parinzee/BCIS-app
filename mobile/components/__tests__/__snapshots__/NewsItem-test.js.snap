// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 10,
      "elevation": 4,
      "height": 150,
      "marginVertical": 20,
      "maxWidth": 700,
      "paddingHorizontal": 15,
      "paddingVertical": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
      "width": 400,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#7e7e7e",
        "borderBottomWidth": 1.25,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      Amazing Title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              undefined,
            ],
          ],
        ]
      }
    >
      üòÅ
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "flex-start",
            "marginBottom": 10,
            "marginTop": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "textAlign": "left",
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.87)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "fontSize": 14,
                    "letterSpacing": 0.25,
                    "lineHeight": 20,
                    "marginVertical": 2,
                  },
                  Object {},
                ],
              ],
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {},
                      Object {},
                    ],
                  ],
                ],
              ]
            }
          >
            Test some 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ],
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "fontWeight": "bold",
                        },
                        Object {},
                      ],
                    ],
                  ],
                ]
              }
            >
              bold
            </Text>
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {},
                      Object {},
                    ],
                  ],
                ],
              ]
            }
          >
            , 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "fontStyle": "italic",
                    },
                  ],
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "fontStyle": "italic",
                        },
                        Object {},
                      ],
                    ],
                  ],
                ]
              }
            >
              italic
            </Text>
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {},
                      Object {},
                    ],
                  ],
                ],
              ]
            }
          >
            , and 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ],
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "fontWeight": "bold",
                        },
                        Object {},
                      ],
                    ],
                  ],
                ]
              }
            >
              both
            </Text>
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {},
                      Object {},
                    ],
                  ],
                ],
              ]
            }
          >
             and 
          </Text>
          <Text
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Object {
                      "color": "#4286f4",
                      "textDecorationLine": "underline",
                    },
                  ],
                ],
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontSize": 14,
                        "letterSpacing": 0.25,
                        "lineHeight": 20,
                        "marginVertical": 2,
                      },
                      Array [
                        Object {
                          "color": "#4286f4",
                          "textDecorationLine": "underline",
                        },
                        Object {},
                      ],
                    ],
                  ],
                ]
              }
            >
              links
            </Text>
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "letterSpacing": 0.25,
                      "lineHeight": 20,
                      "marginVertical": 2,
                    },
                    Array [
                      Object {},
                      Object {},
                    ],
                  ],
                ],
              ]
            }
          >
            !
          </Text>
        </Text>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 12,
                "letterSpacing": 0.4,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              Object {
                "flex": 0,
                "textAlign": "right",
              },
            ],
          ],
        ]
      }
    >
      Posted
       
      17/6/2022
    </Text>
  </View>
</View>
`;
